packages/block-library/src/table-of-contents/save.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Internal dependencies
+ */
+import * as Utils from './utils';
+import ListLevel from './ListLevel';
+
+export default function save( props ) {
+	const { attributes, setAttributes } = props;
+	const headings = attributes.headings;
+
+	if ( headings.length === 0 ) {
+		return null;
+	}
+
+	Utils.updateHeadingBlockAnchors();
```

Comment:
I really like this suggestion, since it removes any guess work and requires explicit user input for entries in the table of contents to become links. I've changed the code to work this way, and it has simplified it a great deal. Now the block does not change any other components.